examples/chef/common/chef-resource-monitoring-delegates.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/clusters/resource-monitoring-server/resource-monitoring-cluster-objects.h>
+#include <app/clusters/resource-monitoring-server/resource-monitoring-server.h>
+#include <chef-resource-monitoring-delegates.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ResourceMonitoring;
+using namespace chip::app::Clusters::ActivatedCarbonFilterMonitoring;
+using namespace chip::app::Clusters::HepaFilterMonitoring;
+using chip::Protocols::InteractionModel::Status;
+
+constexpr std::bitset<4> gHepaFilterFeatureMap{ static_cast<uint32_t>(ResourceMonitoring::Feature::kCondition) |
```

Comment:
if you use `chip::BitMask<ResourceMonitoring::Feature>` then you do not need as many casts. you can have:

```cpp
const chip::BitMask<ResourceMonitoring::Feature> gHepFilterFeatureMap(
    ResourceMonitoring::Feature::kCondition,
    ResourceMonitoring::Feature::kWarning,
    ResourceMonitoring::Feature::kReplacementProductList
);
```

and later use `gHepFilterFeatureMap.Raw()` for the uint32_t value if you actually need it.

Downside seems to not be constexpr, however maybe we can make those methods constexpr at some point if that is a problem.

Reply:
created #31343 to make bitmask constructors constexpr. Bitflags already were.